knowledge-base/treelist-expand-nodes-programmatically.md

```diff
@@ -0,0 +1,187 @@
+---
+title: How to Programmatically Expand or Collapse TreeList Nodes
+description: Learn how to programmatically expand or collapse nodes in a Telerik TreeList for Blazor by utilizing the TreeList state management.
+type: how-to
+page_title: How to Programmatically Expand or Collapse TreeList Nodes
+slug: treelist-kb-expand-nodes-programmatically
+tags: treelist, expand, collapse
+res_type: kb
+ticketid: 1663716, 1649445, 1548181
+---
+
+## Environment
+
+<table>
+<tbody>
+<tr>
+<td>Product</td>
+<td>TreeList for Blazor</td>
+</tr>
+</tbody>
+</table>
+
+## Description
+
+In the Telerik TreeList for Blazor, I load the TreeList in a collapsed mode. I need to programmatically expand and collapse a specific node programmatically.
+
+This KB article also answers the following questions:
+- How can I control the expanded state of TreeList nodes in code?
+- Is it possible to expand a TreeList node without user interaction?
+- What approach should I take to programmatically adjust the visibility of TreeList nodes?
+
+## Solution
+
+To control the expanded or collapsed state of items in the TreeList programmatically, utilize the TreeList State. The state management feature allows for the adjustment of item visibility through code.
+
+Below is an example demonstrating how to use the TreeList State to expand or collapse items programmatically:
+
+````CSHTML
+@using Telerik.DataSource;
+
+<div>
+    <span>
+        <TelerikButton OnClick="@SetTreeListExpandedItems">Expand/Collapse All Items</TelerikButton>
+    </span>
+</div>
+<br />
+<TelerikTreeList Data="@Data"
+                 ItemsField="@(nameof(Employee.DirectReports))"
+                 Reorderable="true"
+                 Resizable="true"
+                 Sortable="true"
+                 FilterMode="@TreeListFilterMode.FilterRow"
+                 Pageable="true"
+                 Width="850px"
+                 OnStateInit="((TreeListStateEventArgs<Employee> args) => OnStateInitHandler(args))"
+                 @ref="TreeListRef">
+    <TreeListColumns>
+        <TreeListColumn Field="Name" Expandable="true" Width="320px" />
+        <TreeListColumn Field="Id" Editable="false" Width="120px" />
+        <TreeListColumn Field="EmailAddress" Width="220px" />
+        <TreeListColumn Field="HireDate" Width="220px" />
+    </TreeListColumns>
+</TelerikTreeList>
+
+@code {
```

Comment:
Consider the [@code Section Order](https://github.com/telerik/blazor/wiki/Docs-%26-Demos-Code-Style#code-section-order)